docs/integrations/logfire.md

```diff
@@ -0,0 +1,164 @@
+# Logfire
+
+[Logfire](https://docs.pydantic.dev/logfire/), a new tool from Pydantic, is built on OpenTelemetry. As Mirascope is also built on Pydantic, it's appropriate for us to ensure seamless integration with them.
+
+## How to use Logfire with Mirascope
+
+```python
+pip install logfire
+
+from mirascope.integrations.logfire import with_logfire
+```
+
+`with_logfire` is a decorator that can be used on all Mirascope classes to automatically log both Mirascope calls and also all our [supported LLM providers](https://docs.mirascope.io/latest/SUPPORTED_PROVIDERS/).
+
+## Examples
+
+### Call
+
+A Mirascope call:
+
+```python
+import logfire
+
+from mirascope.core import anthropic, prompt_template
+from mirascope.integrations.logfire import with_logfire
+
+logfire.configure()
+
+def format_book(title: str, author: str):
+    return f"{title} by {author}"
+
+@with_logfire
+@anthropic.call(model="claude-3-5-sonnet-20240620", tools=[format_book])
+@prompt_template("""Recommend a {genre} book.""")
+def recommend_book(genre: str): ...
+
+recommend_book("fantasy")
+```
+
+This will give you:
+
+* A span called `recommend_book` that captures items like the prompt template, templating properties and fields, and input/output attributes
+* Human-readable display of the conversation with the agent
+* Details of the response, including the number of tokens used
+
+![logfire-call](../assets/logfire-call.png)
+
+### Stream
+
+Streaming a Mirascope call:
+
+```python
+import logfire
+
+from mirascope.core import openai, prompt_template
+from mirascope.integrations.logfire import with_logfire
+
+logfire.configure()
+
+@with_logfire
+@openai.call(
+    model="gpt-4o-mini",
+    stream=True,
+    call_params={"stream_options": {"include_usage": True}},
+)
+@prompt_template("""Recommend a {genre} book.""")
+def recommend_book(genre: str): ...
+
+
+for chunk, _ in recommend_book("fantasy"):
+    print(chunk.content, end="", flush=True)
+```
+
+For some providers, certain `call_params` will need to be set in order for usage to be tracked.
+Also note that the span will not be logged until the stream has been exhausted.
+
+### Response Model
+
+Since Mirascope `response_model` is built on top of [Pydantic](https://docs.pydantic.dev/latest/), you can use the [Pydantic Plugin](https://docs.pydantic.dev/latest/concepts/plugins/) to track additional logs and metrics about model validation, which you can enable using the pydantic_plugin configuration.
+
+This can be particularly useful when [extracting structured information using LLMs](https://docs.mirascope.io/latest/learn/response_models/):
```

Comment:
for now for internal links to our own docs just write (ADD LINK) since these may change. External links are fine (like the ones you have for Pydantic).

Reply:
done